core/drivers/pl022_spi.c

```diff
@@ -273,10 +265,11 @@ static void pl022_sanity_check(struct pl022_data *pd)
 {
 	assert(pd);
 	assert(pd->chip.ops);
-	assert(pd->gpio);
-	assert(pd->gpio->ops);
-	assert(pd->base);
-	assert(pd->cs_gpio_base);
+	/* pd->cs_data.cs_cb CAN be NULL so no assert */
+	if (!pd->b_cs_callback) {
```

Comment:
What about proposing a default callback for gpio based CS ?
This would make the code always go though a callback.
If my proposal simplifies the registration and executive sequence, it's fine. If it adds complexity, just forget it.